### PR TITLE
SC : add GetDecodedAnchroingTx API

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR introduces GetDEcodedAnchroingTx API like below.

```
> klay.getDecodedAnchoringTransactionByHash("0x7e5dab23ba799b24f478077f9fe1478743c7643860de14f1921cfe083e955d81")
{
  BlockHash: "0x3d7ffd4a26a96668c964a4bd826dfe30b54946e9a3ceee771e05d477d776ac02",
  BlockNumber: 598,
  ParentHash: "0x599dc91ca0fc893785f0d398fdfc89900894ff29864f81669591680a2b95de68",
  Period: 1,
  ReceiptHash: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  StateRootHash: "0x15c89c9ba37e1c7620e5981087e312d7cef1e2c5f3fe300d307fb72d7614996a",
  TxCount: 0,
  TxHash: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
}
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: api/api_public_transaction_pool.go [hunk 1]
```diff
@@ -149,6 +150,55 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
...
+		goto decode
+	}
+
+	// No finalized transaction, try to retrieve it from the pool
+	if tx = s.b.GetPoolTransaction(hash); tx != nil {
+		goto decode
+	}
+	return nil, errors.New("can't find the transaction")
+
+decode:
```
Comment (aidan-kwon) [inline api/api_public_transaction_pool.go]: Why don't you use `types.DecodeAnchoringData`?

File: console/jsre/deps/web3.js [hunk 2]
```diff
@@ -5443,6 +5443,12 @@ var methods = function () {
         outputFormatter: formatters.outputTransactionFormatter
     });
 
+    var getDecodedAnchoringTransactionByHash = new Method({
+      name: 'getDecodedAnchoringTransactionByHash',
```
Comment (aidan-kwon) [inline console/jsre/deps/web3.js]:
```suggestion
         name: 'getDecodedAnchoringTransactionByHash',
```

File: api/api_public_transaction_pool.go (hunk 1 above)
Comment (KimKyungup) [reply api/api_public_transaction_pool.go]: Thanks. @aidan-kwon I reflected. :)

File: api/api_public_transaction_pool.go [hunk 3]
```diff
@@ -149,6 +150,48 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
...
+// GetDecodedAnchoringTransactionByHash returns the decoded anchoring data of anchoring transaction for the given hash
+func (s *PublicTransactionPoolAPI) GetDecodedAnchoringTransactionByHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	var tx *types.Transaction
+
+	// Try to return an already finalized transaction
+	if tx, _, _, _ = s.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
+		goto decode
+	}
+
+	// No finalized transaction, try to retrieve it from the pool
```
Comment (jeonghonah) [inline api/api_public_transaction_pool.go]: Is it really necessary? I think that checking chainDB is enough.

Comment (KimKyungup) [reply api/api_public_transaction_pool.go]: I just copied this code from `GetTransactionByHash` at line:154. :)
I think this API is similar with `GetTransactionByHash` API. So I think it's ok. What do you think?

File: console/jsre/deps/web3.js (hunk 2 above)
Comment (KimKyungup) [reply console/jsre/deps/web3.js]: reflected :)

File: api/api_public_transaction_pool.go [hunk 4]
```diff
@@ -149,6 +150,48 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
...
+	if tx = s.b.GetPoolTransaction(hash); tx != nil {
+		goto decode
```
Comment (aidan-kwon) [inline api/api_public_transaction_pool.go]: Why don't you change if conditions to `tx == nil` and then `return nil, errors`?
In that case, we can remove goto statement which may help to make spaghetti code later

Comment (KimKyungup) [reply api/api_public_transaction_pool.go]: I aligned this code with `GetTransactionByHash`. 
There are 2 cases that can be decoded. So If this if statement is changed as you said, we can't remove `goto`.
If we add 1 more case that can be decoded, I think this code is better.

**Pull request merged**